### ISSUE TITLE
Updates and ownership

### ISSUE THREAD

**Issue by tombruijn** [brianhempel/mongo_session_store#34]
Hi there!

I was looking at this gem and noticed it hasn't been updated in a while. @brianhempel are you still supporting it?

My interest in this gem comes from the need for the feature @AppSignal. I updated the gem in my own fork https://github.com/appsignal/mongo_session_store (~~branch [update](https://github.com/tombruijn/mongo_session_store/tree/update)~~), so we can use an updated version, but we can also help out with updating this gem or even take over maintenance. Not promising any long time support or anything outside the scope of our needs, but these changes might be useful for some in a new version of the gem.

In my fork I've updated the gem to use more recent versions of mongo, mongoid, rack and RSpec. No Rails 5 support yet, since we're not using it at this time. I removed support for mongo_mapper and jruby, to make the upgrade easier, but this functionality could easily be restored in the future.

I'm planning on expanding the test suite to test more parts individually and to add the feature of a custom session class, which might just avoid the whole different implementations problem (mongo, mongoid, mongo_mapper).

Let me know what you think @brianhempel !

**Comment by drgcms** [comment]
It looks like the repository is abandoned. I also am interested in further development of this gem. I made my own fork where I made updates necessary for rails 5 compatibility. My first test were positive and my application seems to work OK with rails 5. 

But I have another idea. If you agree and I think it would be better if there is more of us, I would like to ask official MongoDB team to take this gem under https://github.com/mongodb repository.

by
TheR

**Comment by brianhempel** [comment]
I'm fine with a MongoDB takeover.

Otherwise, ping this thread again in a month and I should have some time then to give this repo some attention/hand off to willing maintainers.

**Comment by tombruijn** [comment]
Not sure if this would really fall under the MongoDB org, which is more focussed on the drivers and ODMs rather than community projects.

Meanwhile, I updated the project in my fork https://github.com/appsignal/mongo_session_store 
I refactored the project considerably and can submit a PR to merge it upstream for a version 7. We'd just need to discuss the things I changed and took out.

**Comment by estolfo** [comment]
Hello all!

I maintain Mongoid and the Ruby driver at MongoDB. Thanks for trusting us to take over this gem = )

Unfortunately we don't have the resources to adopt this project right now but I would recommend instead proposing that it be added to the mongoid organization. The mongoid organization already has a bunch of mongoid-related projects in it. See these instructions: http://mongoid.github.io/misc/adding

cc @dblock

**Comment by dblock** [comment]
We would be happy to see any mongoid project moved into the mongoid org. The most important people we need to agree are the gem owners, @nmerouze and @did - we should transfer the topmost repo and start merging commits on that, or transfer any active fork. Can either of you comment?

**Comment by brianhempel** [comment]
Are there any MongoMapper users still out there? Presumably moving into the Mongoid org would mean dropping MM support.I would hope that the plain MongoDB driver could still be supported so you could use it without an ORM.

**Comment by nmerouze** [comment]
I'm okay moving it to another org.

**Comment by tombruijn** [comment]
> Are there any MongoMapper users still out there? Presumably moving into the Mongoid org would mean dropping MM support.

MongoMapper doesn't seem that actively developed anymore. Look like they were working on a 0.14 release back in March: https://github.com/mongomapper/mongomapper But not much happened since. If people are still actively using it we can always add support back at a later time.

> In any case, I would hope that the plain MongoDB driver could still be supported so you could use it without an ORM.

Yes, I think so too. It's not a lot of trouble to support multiple storage implementations.

---

If the Mongoid team agrees, let's move it to the Mongoid organization. We should probably move this repository as it was the most recently maintained and has the most stars. We could always ask GitHub to unlink it as a fork (because submitting PRs on the same repo when it's a fork defaults to a PR to the upstream repository, which is annoying).

Let me know how I can help. I'm happy to help out and maintain it.

**Comment by estolfo** [comment]
@brianhempel You can always use the mongo gem, which is the official MongoDB Ruby driver if you need a driver without an ODM.

**Comment by dblock** [comment]
@nmerouze Excellent! Move the parent repo to me and I'll move it into mongoid. Give rubygems permissions to the mongoid org maintainers, http://mongoid.github.io/misc/adding. I'm here to help.

**Comment by nmerouze** [comment]
@dblock I've done the transfer and added mongoid org maintainers to the gem.

**Comment by dblock** [comment]
Thanks. The repo is now https://github.com/mongoid/mongo_session_store. Same owners, feel free to PR/merge/admin/whatever. I'll try to help a bit.

**Comment by tombruijn** [comment]
Alright. So what do we do? I've submitted a PR (https://github.com/mongoid/mongo_session_store/pull/13) to update the repo in the Mongoid organization with the updates from the last 6 years. Shall we work towards a version 3 for the mongo_session_store gem?

**Comment by tombruijn** [comment]
Thanks everyone for their work! 🎉  I merged all the changes from over the years in [PR #13](https://github.com/mongoid/mongo_session_store/pull/13) along with more recent updates.

Repo has been moved and disconnected from the upper fork tree, making the mongoid one the root upstream repo for many forks. Feel free to contribute!

I've created issues for remaining work and discussions for a new version. I'll work towards a version 3 (following the mongo_session_store gem's release versioning) and release it soon.
